app/services/first_registration/prospective_service.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module FirstRegistration
+  class ProspectiveService
+    def initialize(prospective)
+      @prospective = prospective
+      @concrete    = prospective.build_concrete_user(user)
```

Comment:
Ben olsam DRY olmak adina yazdigim iki duruma dusmek yerine `initialize_employee` ve `initialize_student` tanimlar ve burada da type check yapardim:

```ruby
      @concrete    = if prospective.is_a?(Employee)
                       prospective.initialize_employee(user)
                     elsif prospective.is_a?(Student)
                       prospective.initialize_student(user)
                     end
```

Reply:
Önceki yorumumla ilişkili olarak...  Polimorfizm burada senin önerdiğin "type checking"den (ki OOP'de bu bir antipaterndir) kurtarıyor.  C2'den okuyalım: http://wiki.c2.com/?SwitchStatementsSmell veya https://refactoring.guru/replace-conditional-with-polymorphism

Başka şekilde ifade edecek olursak "tell, don't ask"  yapıyoruz: http://rubyblog.pro/2016/09/tell-dont-ask-principle

Reply:
Bu arada...  @msdundar review için teşekkürler, en yoğun günlerinde zaman ayırıp yazdığın için.  İrtibatı koparmayalım :+1:

Reply:
iki yorumunuza da katiliyorum hocam, tesekkurler 👍